File: app_pojavlauncher/src/main/java/net/kdt/pojavlaunch/modloaders/modpacks/models/ModItem.java
package net.kdt.pojavlaunch.modloaders.modpacks.models;

import androidx.annotation.NonNull;

public class ModItem extends ModSource {

    public final boolean isModpack;
    public final String title;
    public final String description;
    public final String imageUrl;

    public ModItem(int apiSource, boolean isModpack, String id, String title, String description, String imageUrl) {
        super(apiSource, id);
        this.isModpack = isModpack;
        this.title = title;
        this.description = description;
        this.imageUrl = imageUrl;
    }

    @NonNull
    @Override
    public String toString() {
        return "ModItem{" +
                "id='" + id + '\'' +
                ", title='" + title + '\'' +
                ", description='" + description + '\'' +
                ", imageUrl='" + imageUrl + '\'' +
                ", apiSource=" + apiSource +
                ", isModpack=" + isModpack +
                '}';
    }
}

File: app_pojavlauncher/src/main/java/net/kdt/pojavlaunch/modloaders/modpacks/models/ModItem.kt
package net.kdt.pojavlaunch.modloaders.modpacks.models

open class ModItem(
    apiSource: Int,
    isModpack: Boolean,
    id: String?,
    title: String?,
    description: String?,
    imageUrl: String?
) : ModSource() {
    var id: String?
    var title: String?
    var description: String?
    var imageUrl: String?

    init {
        this.apiSource = apiSource
        this.isModpack = isModpack
        this.id = id
        this.title = title
        this.description = description
        this.imageUrl = imageUrl
    }

    override fun toString(): String {
        return "ModItem{" +
                "id='" + id + '\'' +
                ", title='" + title + '\'' +
                ", description='" + description + '\'' +
                ", imageUrl='" + imageUrl + '\'' +
                ", apiSource=" + apiSource +
                ", isModpack=" + isModpack +
                '}'
    }

    val iconCacheTag: String
        get() = apiSource.toString() + "_" + id
}
